model/src/main/kotlin/ScannerRun.kt

```diff
@@ -273,6 +273,50 @@ data class ScannerRun(
         scanResultsById.mapValues { (_, scanResults) ->
             scanResults.flatMapTo(mutableSetOf()) { it.summary.issues }
         }.zipWithSets(issues)
+
+    operator fun plus(other: ScannerRun): ScannerRun {
+        require(environment == other.environment) {
+            "Cannot merge ScannerRuns with different environments: $environment != ${other.environment}."
+        }
+
+        require(config == other.config) {
+            "Cannot merge ScannerRuns with different configurations: $config != ${other.config}."
+        }
+
+        val mergedFiles = (files.toList() + other.files.toList())
```

Comment:
Is it really necessary to convert there to lists? The iteration order of sets should be maintained.

Reply:
It is, it surfaces with the `ScannerIntegrationFunTest` / `return the expected (merged) file lists` in the upcoming PR.

I'll add a small unit test that captures the difference in this PR.